packages/ui/src/components/va-button/VaButton.vue

```diff
@@ -1,241 +1,211 @@
-    })
-
-    const computedClass = computed(() => ({
-      'va-button--default': !props.flat && !props.outline && !props.disabled,
-      'va-button--flat': props.flat,
-      'va-button--outline': props.outline,
-      'va-button--disabled': props.disabled,
-      'va-button--hover': hoverState.value,
-      'va-button--focus': focusState.value,
-      'va-button--large': props.size === 'large',
-      'va-button--small': props.size === 'small',
-      'va-button--normal': !props.size || props.size === 'medium',
-      'va-button--loading': props.loading,
-      'va-button--block': props.block,
-      'va-button--square': !props.rounded,
-      'va-button--round': props.round || (!slots.default && hasOneIcon.value),
-      'va-button--no-label': !isSlotContentPassed.value,
-      'va-button--space-between-items': props.spaceBetweenItems,
-    }))
-
-    const loaderSize = computed(() => {
+    // loader size
+    const { sizeComputed } = useSize(props)
+    const loaderSizeComputed = computed(() => {
       const size = /([0-9]*)(px)/.exec(sizeComputed.value) as null | [string, string, string]
-
       return size ? `${+size[1] / 2}${size[2]}` : sizeComputed.value
     })
 
-    const computedStyle: ComputedRef<Partial<CSSStyleDeclaration>> = computed(() => {
-      const borderColor = props.outline ? colorComputed.value : ''
-
-      let background = props.gradient
-        ? getGradientBackground(colorComputed.value)
-        : colorComputed.value
+    // attributes
+    const { tagComputed } = useRouterLink(props)
+    const attributesComputed = useButtonAttributes(props)
 
-      if (isTransparentBackground.value) {
-        background = 'var(--va-transparent)'
-      }
+    // states
+    const button = shallowRef<HTMLElement>()
+    const { isFocused, focus, blur } = useFocus(button)
+    const { isHovered } = useHover(button)
+    const { isPressed } = usePressed(button)
+
+    // icon attributes
+    const iconColorComputed = computed(() => props.iconColor ? getColor(props.iconColor) : textColorComputed.value)
+    const iconAttributesComputed = computed(() => ({
+      size: props.size,
+      color: iconColorComputed.value,
+    }))
 
-      if (hoverState.value) {
-        const alpha = props.outline ? -0.9 : -0.8
-        const lightness = 5
-        const color = isTransparentBackground.value
-          ? shiftHSLAColor(colorComputed.value, { a: alpha })
-          : shiftHSLAColor(colorComputed.value, { l: lightness })
+    // classes
+    const wrapperClassComputed = computed(() => ({ 'va-button__content--loading': props.loading }))
 
-        background = props.gradient ? getGradientBackground(color) : color
-      }
+    const isSlotContentPassed = computed(() => !!slots.default?.()?.[0]?.children)
+    const isOneIcon = computed(() => !!((props.iconRight && !props.icon) || (!props.iconRight && props.icon)))
+    const computedClass = useBem('va-button', () => ({
```

Comment:
I would suggest  keeping full bem classes if it's not too much pain.
The upside is that keeps them easily searchable. And from what I perceive it doesn't do that much otimization-wise.